tests/unit_tests/c_wrappers/job_runner/test_jobmanager.py

```diff
@@ -187,4 +183,85 @@ def test_exec_env():
             with open("exec_env_exec_env.json", encoding="utf-8") as f:
                 exec_env = json.load(f)
                 assert exec_env["TEST_ENV"] == "123"
-                assert exec_env["NOT_SET"] is None
+        if isinstance(msg, Exited):
+            assert msg.exit_code == 0
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_job_env():
+    with open("run_me.py", "w", encoding="utf-8") as f:
+        f.write(
+            """#!/usr/bin/env python\n
+import os
+assert os.environ["TEST_ENV"] == "123"
+            """
+        )
+    os.chmod("run_me.py", stat.S_IEXEC + stat.S_IREAD)
+
+    with open("RUN_ENV", "w", encoding="utf-8") as f:
+        f.write("EXECUTABLE run_me.py\n")
+        f.write("ENV TEST_ENV 123\n")
+
+    job = ExtJob.from_config_file(name=None, config_file="RUN_ENV")
+    forward_model = [job]
+
+    with open("jobs.json", mode="w", encoding="utf-8") as fptr:
+        json.dump(ErtConfig.forward_model_data_to_json(forward_model, "run_id"), fptr)
+
+    with open("jobs.json", "r", encoding="utf-8") as f:
+        jobs_json = json.load(f)
+
+    # Check ENV variable not available outside of job context
+    assert "TEST_ENV" not in os.environ
+
+    for msg in list(JobRunner(jobs_json).run([])):
+        if isinstance(msg, Exited):
+            assert msg.exit_code == 0
+
+    # Check ENV variable not available outside of job context
+    assert "TEST_ENV" not in os.environ
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_job_default_env():
+    with open("run_me.py", "w", encoding="utf-8") as f:
+        f.write(
+            """#!/usr/bin/env python\n
+import os
+assert os.environ["_ERT_ITERATION_NUMBER"] == "0"
+assert os.environ["_ERT_REALIZATION_NUMBER"] == "0"
+assert os.environ["_ERT_RUNPATH"] == "./"
+            """
+        )
+    os.chmod("run_me.py", stat.S_IEXEC + stat.S_IREAD)
+
+    with open("RUN_ENV", "w", encoding="utf-8") as f:
+        f.write("EXECUTABLE run_me.py\n")
+
+    job = ExtJob.from_config_file(name=None, config_file="RUN_ENV")
+    forward_model = [job]
+
+    with open("jobs.json", mode="w", encoding="utf-8") as fptr:
+        json.dump(
+            ErtConfig.forward_model_data_to_json(
+                forward_model,
+                "run_id",
+                context=SubstitutionList.from_dict({"DEFINE": [["<RUNPATH>", "./"]]}),
+            ),
+            fptr,
+        )
+
+    with open("jobs.json", "r", encoding="utf-8") as f:
+        jobs_json = json.load(f)
+
+    # Check default ENV variable not available outside of job context
+    for k in ExtJob.default_env:
+        assert k not in os.environ
+
+    for msg in list(JobRunner(jobs_json).run([])):
+        if isinstance(msg, Exited):
+            assert msg.exit_code == 0
+
+    # Check default ENV variable not available outside of job context
+    for k in ExtJob.default_env:
+        assert k not in os.environ
```

Comment:
Am i correct in  that there is a hidden verification in here, -- you indirectly check that the default environment set in the python script is present, and after jobs have finished, you verify that the os.environment is not polluted?

Reply:
Yes, that is correct.